apps/sms/views/inbox/js/inbox.js

```diff
@@ -622,18 +551,24 @@ var InboxView = {
           if (!Threads.has(draft.id)) {
             var thread = Thread.create(draft);
             Threads.set(draft.id, thread);
-            this.appendThread(thread);
+            promises.push(this.appendThread(thread));
           }
         }
       }
 
-      this.sticky && this.sticky.refresh();
+      return Promise.all(promises);
     });
   },
 
   prepareRendering: function inbox_prepareRendering() {
-    this.container.innerHTML = '';
-    this.renderDrafts();
+    // this.container.innerHTML = '';
+    this.listData = [];
+    this.list.configure({
+      getSectionName: function(item) {
+        return item.dayTextContent;
+      }
+    });
+    return this.prepareDrafts();
```

Comment:
if drafts are ready after the list started to render, isn't it possible to update the list and ask for a rerender ?